lib/pbench/server/api/resources/query_apis/query_dataset.py

```diff
@@ -0,0 +1,268 @@
+from flask import request, jsonify
+from flask_restful import Resource, abort
+import requests
+
+from dateutil import parser
+from pbench.server.api.resources.query_apis import (
+    get_es_url,
+    get_index_prefix,
+    gen_month_range,
+    get_user_term,
+)
+
+
+class QueryDataset(Resource):
+    """
+    Abstracted Pbench API to get detailed data for a particular
+    dataset by name. The date range given isn't used for the search
+    directly, but rather allows building a set of Elasticsearch index
+    names that will contain the data. (Note that the dashboard uses the
+    same date range it used to find controllers; however it could make this
+    more efficient by using only the year and month in which the result
+    was generated.)
+
+    Note that although the name "queryResult" is retained for consistency
+    with the existing dashboard UI code, the word "result" is a bit
+    misleading in that this deals with Pbench "run" documents, not Pbench
+    "results" documents. It's a follow-on to queryResults, which returns
+    the list of datasets within a specified date range for a given controller
+    (as reported by queryControllers). The queryResult API returns details from
+    the dataset run document; basically the union of the "run" and "@metadata"
+    sub-documents.
+    """
+
+    def __init__(self, config, logger):
+        self.logger = logger
+        self.es_url = get_es_url(config)
+        self.prefix = get_index_prefix(config)
+
+    def post(self):
+        """
+        POST to get details for a specific Pbench dataset within the
+        specified date range, and owned by a specified user:
+
+        {
+            "user": "email-tag",
```

Comment:
Interesting, this could be a username as well right?

Reply:
The comment is perhaps stale, although I'm not sure we've entirely resolved the issue.

The dashboard has never had a "username", only "email". You coded our identity manager with the more conventional username + email style, but I'd written this comment very early on and followed the email convention used in the dashboard.

We really ought to discuss exactly what we're going to do; what user metadata we maintain along with the primary access key, conventions for updating email, etc. For now, yeah, this is whatever token is used to uniquely reference a specific user ... or `None` for public data.